entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt

```diff
@@ -134,6 +135,28 @@ class Neo4jRepository(
         return session.query(query, parameters, true).toList().isNotEmpty()
     }
 
+    fun hasPropertyInstance(subjectNodeInfo: SubjectNodeInfo, propertyName: String, datasetId: String?): Boolean {
+        val query = if (datasetId == null)
+            """
+            MATCH (a:${subjectNodeInfo.label} { id: ${'$'}attributeId })-[:HAS_VALUE]->(property:Property { name: ${'$'}propertyName })
+            WHERE NOT EXISTS (property.datasetId)

```

Comment:
How do you access default property ? This is for the old properties ? Shouldn't we migrate what's already in the DB ?

Reply:
The default property is the property having no datasetId (we can say that all the existing properties in the database are default).
To update the default property the update fragment should not contain a datasetId.
Yes this works for the old properties.
No data migration is needed.